### PR TITLE
Add codeactions related to switch-Statements

### PR DESCRIPTION
**Pull request by JCWasmx86**
- AddDefaultToSwitchCodeAction (Add default case to switch)
- AddOtherConstantsToSwitchAction (Add missing enum constants to switch)

### CODE REVIEW

File: src/codehelp/nodesearch.vala [hunk 1]
```diff
@@ -595,7 +595,7 @@ class Vls.NodeSearch : Vala.CodeVisitor {
 
     public override void visit_switch_statement (Vala.SwitchStatement stmt) {
         if (!seen.add (stmt)) return;
-        if (this.match (stmt))
+        if (this.match (stmt) || new Range.from_sourceref (stmt.source_reference).contains (pos))
```
Comment (Prince781) [inline src/codehelp/nodesearch.vala:598]: See if you can update `match()` instead. I'd prefer to have all the logic in one place.

Comment (JCWasmx86) [reply src/codehelp/nodesearch.vala:598]: Just do something like this in match:

```vala
... match(...) {
  ...
  if (node is Vala.SwitchStatement) {
    return new Range.from_sourceref (( Vala.SwitchStatement)node).source_reference).contains (pos)
  }
}
```

**Comment by Prince781** [comment]
I updated the commit just now with an indentation API for `CodeStyleAnalyzer`. It's still a work-in-progress.

**Comment by Prince781** [comment]
Once I improve the indentation API to look at surrounding statements, then refactor your change around this, I'll merge this PR.

**Comment by Prince781** [comment]
I still have to fix the case when dealing with empty switch statements. Then I'll merge.

**Comment by Prince781** [comment]
Just one thing left: to disable code actions when the document is in an inconsistent state.

**Comment by Prince781** [comment]
Also, we want these code actions to be presented as fixes for a diagnostic.

**Comment by Prince781** [comment]
@JCWasmx86 thank you!

**Pull request merged**